### ISSUE TITLE
CMake GLEW handling on Windows is broken-ish

### ISSUE THREAD

**Issue by jcowles** [PixarAnimationStudios/USD#53]
On Windows, when statically linking against Glew, GLEW_STATIC must be defined, here is the relevant part of glew.h:

```
/*
 * GLEW_STATIC is defined for static library.
 * GLEW_BUILD  is defined for building the DLL library.
 */

#ifdef GLEW_STATIC
#  define GLEWAPI extern
#else
#  ifdef GLEW_BUILD
#    define GLEWAPI extern __declspec(dllexport)
#  else
#    define GLEWAPI extern __declspec(dllimport)
#  endif
#endif
```

So on Windows when using glew32.lib (or glew32d.lib, which OpenSubdiv/USD doesn't support!) -> no switch needed.

When using glew32s.lib (glew32sd.lib) -> -DGLEW_STATIC is required.

Also, OpenSubdiv should guard against glew leaking into the public headers/templates, which I think is what I ran into -- will file a separate bug.

**Comment by jcowles** [comment]
I have a fix

**Comment by jtran56** [comment]
Filed as internal issue #137371.

**Comment by sunyab** [comment]
We think this issue should be fixed in OpenSubdiv 3.1.1 by this commit:
PixarAnimationStudios/OpenSubdiv#905

I'm going to close this (and the associated pull request) out for now, please reopen if this is still an issue. Thanks!
